File: shuttle-api/src/ShuttleApi.Domain/Drivers/DriverRosterEntry.cs
using ShuttleApi.Domain.Common;

namespace ShuttleApi.Domain.Drivers;

public sealed class DriverRosterEntry : Entity<Guid>
{
    public Guid DriverId { get; private set; }
    public DateOnly EntryDate { get; private set; }
    public RosterStatus Status { get; private set; }
    public TimeOnly? ShiftStart { get; private set; }
    public TimeOnly? ShiftEnd { get; private set; }
    public DateTime UpdatedAt { get; private set; }

    private DriverRosterEntry() { }

    public static DriverRosterEntry Create(
        Guid driverId,
        DateOnly entryDate,
        RosterStatus status,
        TimeOnly? shiftStart,
        TimeOnly? shiftEnd)
    {
        return new DriverRosterEntry
        {
            Id = Guid.NewGuid(),
            DriverId = driverId,
            EntryDate = entryDate,
            Status = status,
            ShiftStart = shiftStart,
            ShiftEnd = shiftEnd,
            UpdatedAt = DateTime.UtcNow
        };
    }

    public void Update(RosterStatus status, TimeOnly? shiftStart, TimeOnly? shiftEnd)
    {
        Status = status;
        ShiftStart = shiftStart;
        ShiftEnd = shiftEnd;
        UpdatedAt = DateTime.UtcNow;
    }
}

File: shuttle_tablet/lib/features/drivers/domain/entities/driver_roster_entry.dart
import 'package:equatable/equatable.dart';

enum RosterStatus { available, unavailable, scheduled, onTrip }

class DriverRosterEntry extends Equatable {
  final String id;
  final String driverId;
  final DateTime entryDate;
  final RosterStatus status;
  final String? shiftStart; // "HH:mm" format
  final String? shiftEnd;   // "HH:mm" format

  const DriverRosterEntry({
    required this.id,
    required this.driverId,
    required this.entryDate,
    required this.status,
    this.shiftStart,
    this.shiftEnd,
  });

  @override
  List<Object?> get props => [id, driverId, entryDate, status, shiftStart, shiftEnd];
}
